src/Stratis.Bitcoin.IntegrationTests/Wallet/SendingRecentlyStakedCoinsBeforeMaturity_Steps.cs

```diff
@@ -0,0 +1,123 @@
+            this.nodeGroupBuilder = new NodeGroupBuilder();
+        }
+
+        protected override void AfterTest()
+        {
+            this.nodeGroupBuilder?.Dispose();
+        }
+
+        private void two_nodes_which_includes_a_proof_of_stake_node_with_a_million_coins()
+        {
+            this.proofOfStakeSteps.GenerateCoins();
+
+            this.proofOfStakeSteps.ProofOfStakeNodeWithCoins.FullNode.WalletManager()
+                .GetSpendableTransactionsInWallet(this.proofOfStakeSteps.PosWallet)
+                .Sum(utxo => utxo.Transaction.Amount)
+                .Should().Be(Money.Coins(OneMillion));
+        }
+
+        private void a_wallet_sends_staked_coins_before_maturity()
+        {
+            this.SenderNode = this.proofOfStakeSteps.ProofOfStakeNodeWithCoins;
+
+            GetWalletHistory(this.SenderNode, this.proofOfStakeSteps.PosWallet)
+                .AccountsHistoryModel
+                .FirstOrDefault()
+                .TransactionsHistory
+                .Where(txn => txn.Type == TransactionItemType.Send).Count().Should().Be(0);
+
+            var transactionResult = this.SenderNode.FullNode.NodeService<WalletController>().BuildTransaction(new BuildTransactionRequest
+            {
+                AccountName = this.proofOfStakeSteps.WalletAccount,
+                AllowUnconfirmed = true,
+                Amount = new Money((int)OneMillion).ToString(),
+                DestinationAddress = GetReceiverUnusedAddressFromWallet(), 
+                FeeType = FeeType.Medium.ToString("D"),
+                Password = this.proofOfStakeSteps.PosWalletPassword,
+                WalletName = this.proofOfStakeSteps.PosWallet,
+                FeeAmount = Money.Satoshis(20000).ToString()
+            });
+
+            var walletTransactionModel = (transactionResult as JsonResult).Value as WalletBuildTransactionModel;
+
+            this.SenderNode.FullNode.NodeService<WalletController>().SendTransaction(new SendTransactionRequest(walletTransactionModel.Hex));
+        }
+
+        private void the_wallet_history_shows_the_transaction_as_sent()
+        {
+            GetWalletHistory(this.SenderNode, this.proofOfStakeSteps.PosWallet)
+                .AccountsHistoryModel
+                .FirstOrDefault()
+                .TransactionsHistory
+                .Where(txn => txn.Type == TransactionItemType.Send).Count().Should().Be(1);
+
+            this.ReceiverNode.FullNode.WalletManager().GetSpendableTransactionsInWallet(WalletName)
+                .Sum(utxo => utxo.Transaction.Amount)
+                .Should().Be(0);
+        }
+
+        private WalletHistoryModel GetWalletHistory(CoreNode node, string walletName)
+        {
+            var walletHistory = node.FullNode.NodeService<WalletController>()
+                .GetHistory(new WalletHistoryRequest { WalletName = walletName }) as JsonResult;
+
+            return walletHistory.Value as WalletHistoryModel;
```

Comment:
happy with a null ref exception here or prefer a cast exception above?

Reply:
Changed to return walletHistory?.Value as WalletHistoryModel;